test/e2e/alertmanager_test.go

```diff
@@ -472,6 +474,188 @@ func TestAlertmanagerAPI(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	// Check read and write access to the Alertmanager API.
+	testAlertmanagerAPIAccess(t)
+}
+
+func testAlertmanagerAPIAccess(t *testing.T) {
+	ctx := context.Background()
+	const (
+		monitoringNamespace = "openshift-monitoring"
+		testNamespace       = "alertmanager-api-e2e-test"
+	)
+	namespaceObj := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testNamespace,
+			Labels: map[string]string{
+				framework.E2eTestLabelName: framework.E2eTestLabelValue,
+			},
+		},
+	}
+	namespaceObj, err := f.KubeClient.CoreV1().Namespaces().Create(ctx, namespaceObj, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		foreground := metav1.DeletePropagationForeground
+		if err = f.KubeClient.CoreV1().Namespaces().Delete(ctx, testNamespace, metav1.DeleteOptions{PropagationPolicy: &foreground}); err != nil {
+			t.Logf("err deleting namespace %s: %v", testNamespace, err)
+		}
+	})
+
+	// Check access to the Alertmanager API based on the reader and writer roles.
+	const (
+		sa = testNamespace + "-sa"
+	)
+	saCleanup, err := f.CreateServiceAccount(testNamespace, sa)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		err = saCleanup()
+		if err != nil {
+			t.Logf("failed to cleanup service account %s: %v", sa, err)
+		}
+	}()
+	var client *framework.PrometheusClient
+	err = framework.Poll(5*time.Second, time.Minute, func() error {
+		token, err := f.GetServiceAccountToken(testNamespace, sa)
```

Comment:
(nit) GetServiceAccountToken already has a retry loop so it might better to move it outside the Poll.